### PR TITLE
25818: Add ability to upload section 1005 file

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As an Education team member, I need the ability to upload a file containing data regarding an institutions compliance with Isakson and Roe section 1005 so that data can be displayed on the Comparison Tool.

## Original issue(s)
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/25818

## Testing done
Testing passes locally

## Screenshots
N/A

## Acceptance criteria
- [x] The file type is "InStateTuitionPolicyUrl".
- [x] The user has the ability to:
a. Import a data file
b. Export a data file
- [x] Accepted import file formats for the data file are:
a. .txt
b. .csv
c. .xls
d. .xlsx
- [x] Valid column delimiters are:
a. ,
b. |
- [x] The accepted imported file has the columns in the following order:
a. Facility Code
b. In-State Tuition Information
- [x] The following columns are required for data to be imported:
a. FacilityCode
b. In-State Tuition Information
- [x] The "In-State Tuition Information" data is included in the institution version generation.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/utilities/institution_builder/in_state_tuition_policy_url_builder.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module InstitutionBuilder
+  class InStateTuitionPolicyUrlBuilder
+    extend Common
+
+    def self.add_in_state_tuition_policy_url(version_id)
```
Comment (shawkey-daniel) [inline app/utilities/institution_builder/in_state_tuition_policy_url_builder.rb]: Can we just call this self.build?

File: db/migrate/20210615232805_add_in_state_tuition_information_to_institutions_archives.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddInStateTuitionInformationToInstitutionsArchives < ActiveRecord::Migration[6.0]
+  def change
+    add_column :institutions_archives, :in_state_tuition_information, :string
```
Comment (shawkey-daniel) [inline db/migrate/20210615232805_add_in_state_tuition_information_to_institutions_archives.rb:3]: Can we just add this line under the line for institutions? I think they go together

File: db/schema.rb [hunk 3]
```diff
@@ -245,6 +245,13 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "in_state_tuition_policy_urls", force: :cascade do |t|
+    t.string "facility_code"
+    t.string "in_state_tuition_information"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
```
Comment (shawkey-daniel) [inline db/schema.rb:252]: I don't see precision on any other timestamps. Is this needed?

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

**Comment by zurbergram** [comment]
Can you add the `in_state_tuition_information` column to `app/serializers/institution_profile_serializer.rb` ?

**Review by LindseySaari — APPROVED** [review]
👍

**Pull request merged**